vdr/didx509/resolver_test.go

```diff
@@ -0,0 +1,121 @@
+
+	rootDID := did.MustParseDID(fmt.Sprintf("did:x509:0:%s:%s::san:otherName:%s", "sha256", sha256Sum(rootCertificate.Raw), otherNameValue))
+
+	t.Run("test nulls", func(t *testing.T) {
+		chain := metadata.X509CertChain
+		metadata.X509CertChain = nil
+		_, _, err := resolver.Resolve(rootDID, &metadata)
+		require.Error(t, err)
+		assert.Equal(t, err.Error(), errors.New("x509 rootCert chain is missing").Error())
+		metadata.X509CertChain = chain
+
+	})
+	t.Run("happy flow", func(t *testing.T) {
+		validator.EXPECT().Validate(gomock.Any()).Return(nil)
+		resolve, documentMetadata, err := resolver.Resolve(rootDID, &metadata)
+		require.NoError(t, err)
+		assert.NotNil(t, resolve)
+		assert.NotNil(t, documentMetadata)
+	})
+	t.Run("happy flow 2", func(t *testing.T) {
+		metadata.X509CertThumbprintS256 = ""
+		validator.EXPECT().Validate(gomock.Any()).Return(nil)
+		resolve, documentMetadata, err := resolver.Resolve(rootDID, &metadata)
+		require.NoError(t, err)
+		assert.NotNil(t, resolve)
+		assert.NotNil(t, documentMetadata)
+		metadata.X509CertThumbprintS256 = sha256Sum(signingCert.Raw)
+	})
+	t.Run("happy flow 2", func(t *testing.T) {
+		metadata.X509CertThumbprint = ""
+		validator.EXPECT().Validate(gomock.Any()).Return(nil)
+		resolve, documentMetadata, err := resolver.Resolve(rootDID, &metadata)
+		require.NoError(t, err)
+		assert.NotNil(t, resolve)
+		assert.NotNil(t, documentMetadata)
+		metadata.X509CertThumbprint = sha1Sum(signingCert.Raw)
+	})
+	t.Run("broken chain", func(t *testing.T) {
+		expectedErr := errors.New("broken chain")
+		validator.EXPECT().Validate(gomock.Any()).Return(expectedErr)
+		_, _, err := resolver.Resolve(rootDID, &metadata)
+		require.Error(t, err)
+		assert.Equal(t, err.Error(), expectedErr.Error())
+	})
+	t.Run("wrong otherName value", func(t *testing.T) {
+		rootDID := did.MustParseDID(fmt.Sprintf("did:x509:0:%s:%s::san:otherName:%s", "sha256", sha256Sum(rootCertificate.Raw), "ANOTHER_BIG_STRING"))
+		_, _, err := resolver.Resolve(rootDID, &metadata)
+		require.Error(t, err)
+		assert.Equal(t, err.Error(), ErrWrongSanOtherName.Error())
+	})
+	t.Run("wrong hash type value", func(t *testing.T) {
+		rootDID := did.MustParseDID(fmt.Sprintf("did:x509:0:%s:%s::san:otherName:%s", "kaas", sha256Sum(rootCertificate.Raw), "ANOTHER_BIG_STRING"))
+		_, _, err := resolver.Resolve(rootDID, &metadata)
+		require.Error(t, err)
+		assert.Equal(t, err.Error(), errors.New("unsupported hash algorithm: kaas").Error())
+	})
+	t.Run("wrong hash value", func(t *testing.T) {
+		rootDID := did.MustParseDID(fmt.Sprintf("did:x509:0:%s:%s::san:otherName:%s", "sha256", "kaas", "ANOTHER_BIG_STRING"))
+		_, _, err := resolver.Resolve(rootDID, &metadata)
+		require.Error(t, err)
+		assert.Equal(t, err.Error(), errors.New("cannot find a certificate with alg: sha256 hash: kaas").Error())
+	})
+	t.Run("wrong DID type", func(t *testing.T) {
+		rootDID := did.MustParseDID("did:kaas:gouda.nl:jonge")
```

Comment:
however funny, better use non-existing/example domain (`example.com`)
```suggestion
		rootDID := did.MustParseDID("did:kaas:example.com:jonge")
```

Reply:
Updated most domains and emails to example.com